src/dbnode/storage/types.go

```diff
@@ -1281,3 +1295,8 @@ type AggregateTilesOptions struct {
 	// TODO: remove once we have metrics type stored in the metadata.
 	HandleCounterResets bool
 }
+
+// AfterNamespaceCreatedFn is the type of function invoked after each namespace is created.
+type AfterNamespaceCreatedFn func (Namespace, GetNamespaceFn) error
```

Comment:
Why is a getter passed in here, wouldn't it be better to pass in the direct reference?

Reply:
This is what I tried initially. The problem was that this callback happens under a lock on `databaseNamespace`, and calling some public `databaseNamespace` method from there causes a reentrant locking attempt that hangs.